### DISCUSSION TITLE
Diff issue

### DISCUSSION THREAD

**Discussion by ellisonleao** [ellisonleao/gruvbox.nvim#239] [Ideas]
Opening this discussion page so we can proper discuss the diff highlights issue. Would love to hear any suggestions

Ref: #235 , #105, #269, #273, #275

**Comment by Niverton** [comment]
When using diff mode, the text is black on a bright background (in dark mode). This is not nice on the eye, and makes large changes in a code base (big DiffAdd blocks especially) hard to read due to the lack of syntax highlighting.
The issue is also present in the original community edition gruvbox, and an issue is also open there https://github.com/gruvbox-community/gruvbox/issues/173
Other themes fix the issue by only using the background color for diff highlights, like catpuccin
https://github.com/catppuccin/nvim/blob/main/lua/catppuccin/groups/syntax.lua#L74-L77

**Comment by thallada** [comment]
Whatever we come up with it should be configurable like in https://github.com/ellisonleao/gruvbox.nvim/pull/228. It seems like some people prefer the current diff colors which are very bright and contrasting at the expense of loosing out on finer syntax highlighting. I'm honestly kind of surprised by this since both Github's and vscode's diff colors not like that. They have subtle background colors with syntax highlighting preserved and were more like what I made in #225. I assumed that would be the desired look of a modern editor, but people clearly did not like that (e.g. https://github.com/ellisonleao/gruvbox.nvim/pull/235#issuecomment-1501190369).

People also commented that the colors in #225 were not gruvbox-y enough. I realize now that I probably should have copied colors from an existing gruvbox theme like https://github.com/jdinhify/vscode-theme-gruvbox/blob/main/themes/gruvbox-light-hard.json instead of coming up with the colors myself since I am not a theme designer. It's also possible that the colors I came up with only looked good in the "hard" variants and looked garbage in other variants and I didn't do enough testing.

**Comment by Niverton** [comment]
A bit of an "up" on this discussion, it's been a bit more than a month since the last comment.
In the precedent issues and PR, some people were enthusiastic about the feature and I didn't see any arguments against having the feature. I feel like the discussion boards are unnoticed by most people unfortunately.

Most of the recent themes I see have this feature, either forced or behind a flag. While each project is different, I think that it's a wanted enough feature in the community to be worth considering.

**Comment by julienvincent** [comment]
And to add my 2c, I would also very much appreciate this landing. I ended up moving away from this theme and using the gruvbox-material one instead, even though in general I prefer the colors here. I would gladly switch back if/when this is resolved.

**Comment by timsofteng** [comment]
+1

**Comment by JuanRaya07** [comment]
+1, something more subtle like [this](https://user-images.githubusercontent.com/2786478/131269942-e34100dd-cbb9-48fe-af31-6e518ce06e9e.png) could be cool. LMK if you want me to open a PR on that.

**Reply by ellisonleao** [reply]
sure @JuanRaya07 feel free to open it!

**Comment by tilupe** [comment]
In my opinion you will never hit everybodys taste with the colors. The most important is the functionality and in a diff it is to see what changed. Not only which part was changed but also what functionality and therefore you need semantic highlighting on the diffs. If the color doesn't suit me I just change it via the great feature of this plugin, where I can make overwrites. 
Not sure if I missed the point of the discussion but I would really love to see this feature in the plugin

**Comment by ellisonleao** [comment]
Ok guys, i had some time to work on this issue today, and I've fixed some highlights for neogit, diffview and fugitive.  IMO it fits well for the common usage and should be fine for everybody. This is already included in #280 which will be the 2.0 version. Here are some screenshots:

Diffview:

![diffview](https://github.com/ellisonleao/gruvbox.nvim/assets/178641/fa7c14bb-1216-4153-8328-0645ec3dbe78)
![image](https://github.com/ellisonleao/gruvbox.nvim/assets/178641/4f335768-709e-42f7-ab4c-f23c74137a2f)


Neogit:

![Screenshot from 2023-09-30 18-26-40](https://github.com/ellisonleao/gruvbox.nvim/assets/178641/5924b0f1-113b-40fd-9e88-5291d9a56a67)
![image](https://github.com/ellisonleao/gruvbox.nvim/assets/178641/69560b69-88f5-4330-8901-19bc8c979f6f)


Fugitive:

![Screenshot from 2023-09-30 18-26-54](https://github.com/ellisonleao/gruvbox.nvim/assets/178641/24148676-c739-44f1-a41d-a390f44a394d)
![image](https://github.com/ellisonleao/gruvbox.nvim/assets/178641/c2d10047-6ceb-4971-9f71-d3285a214980)


I am pushing this today so feel free to test and let me know if we close this long time discussion. We will be evolving some specific highlights later (Neogit and their confusing highlights for example)

**Comment by eeeXun** [comment]
The `DiffText` is hard to recognize

![image](https://github.com/ellisonleao/gruvbox.nvim/assets/58657914/dd522300-e1cc-49d7-8bce-daa3f1c82909)

**Reply by ellisonleao** [reply]
This was the only one that I couldn't find a good solution yet, you can try to override with a custom color until I find a better solution

**Reply by ellisonleao** [reply]
did a modification here, which make things a little better for now https://github.com/ellisonleao/gruvbox.nvim/commit/a569160337aa309db4cfe21ba4e088f0e6676fb4

**Comment by ellisonleao** [comment]
folks can we close this one?

**Comment by mrvon** [comment]
can we use old style diff style with options? I just don't like new subtle one.

**Comment by timsofteng** [comment]
@ellisonleao thank you! It works for me
